consensus/common/src/main/java/org/hyperledger/besu/consensus/common/ScheduableProtocolContext.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.common;
+
+import org.hyperledger.besu.consensus.common.bft.BftForksSchedule;
+import org.hyperledger.besu.ethereum.ConsensusContext;
+import org.hyperledger.besu.ethereum.ConsensusContextFactory;
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.chain.GenesisState;
+import org.hyperledger.besu.ethereum.chain.MutableBlockchain;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+import org.hyperledger.besu.ethereum.worldstate.WorldStateArchive;
+
+public class ScheduableProtocolContext extends ProtocolContext {
+
+  private final BftForksSchedule<ConsensusContext> consensusContextSchedule;
+
+  public ScheduableProtocolContext(
+      final MutableBlockchain blockchain,
+      final WorldStateArchive worldStateArchive,
+      final BftForksSchedule<ConsensusContext> consensusContextSchedule) {
+    super(blockchain, worldStateArchive, null);
+    this.consensusContextSchedule = consensusContextSchedule;
+  }
+
+  public static ProtocolContext init(
+      final MutableBlockchain blockchain,
+      final WorldStateArchive worldStateArchive,
+      final GenesisState genesisState,
+      final ProtocolSchedule protocolSchedule,
+      final ConsensusContextFactory consensusContextFactory) {
+    if (blockchain.getChainHeadBlockNumber() < 1) {
+      genesisState.writeStateTo(worldStateArchive.getMutable());
+    }
```

Comment:
Is this actually related to the ProtocolContext? I'm wondering why this is here and not in BesuController?

I see it's come from ProtocolContext.init and so my query is not really relevant to this PR, although arguably this is a good moment to resolve it if we're about to duplicate it.

Reply:
This also wants coverage according to Sonarcloud...not sure if we've agreed to adhere to a sensible coverage limit yet though

Reply:
Don't know why it's there don't seem a bit weird now I think about it. But yes copied it from the ProtocolContext.init method.